### ISSUE TITLE
class not found exception

### ISSUE THREAD

**Issue by lgrapenthin** [venantius/yagni#30]
Relevant parts in stacktrace

java.lang.ClassNotFoundException: my-project.functional.>
...
	at yagni.jvm$is_class_generator_QMARK_.invoke(jvm.clj:45)

my-project.functional is a cljc namespace looking like this:

```clojure
(ns my-project.functional
  #?(:cljs (:require-macros my-project.functional))
  #?(:clj (:refer-clojure :rename {-> core->
                                   ->> core->>
                                   some-> core-some->
                                   some->> core-some->>})))

#?(:clj (defmacro -> [& forms]
          `(fn [x#]
             (core-> x# ~@forms))))

#?(:clj (defmacro ->> [& forms]
          `(fn [x#]
             (core->> x# ~@forms))))

#?(:clj (defmacro some-> [& forms]
          `(fn [x#]
             (core-some-> x# ~@forms))))

#?(:clj (defmacro some->> [& forms]
          `(fn [x#]
             (core-some->> x# ~@forms))))
```

**Comment by venantius** [comment]
Could you post the full stacktrace?

**Comment by lgrapenthin** [comment]
Looking at jvm.clj:45 I'd assume that your code thinks -> is a record constructor. Do you really need any more?

**Comment by venantius** [comment]
It's not an unreasonable request. You're probably right, but I'd rather have the full trace available than not.

**Comment by marcomorain** [comment]
I'm getting the same error – a function named `->epoch-millis` in ns `foo.bar` gives a class not found exception `foo.bar.epoch-millis`.

**Comment by venantius** [comment]
I'll give this a look tonight. Seems like it's probably a fairly straightforward fix.

**Comment by venantius** [comment]
A fix for this is now on the `0.1.3` branch. Problem was replicable and I think this should do the trick.

**Comment by marcomorain** [comment]
@venantius that fixed the problem, thanks.
